H/HDF5/build_tarballs.jl

```diff
@@ -136,6 +78,9 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("Zlib_jll"),
+    Dependency("OpenSSL_jll"),
+    Dependency("LibCURL_jll"),
```

Comment:
Why these?

Reply:
hdf5 depends on them.

Reply:
Why we never needed them?

Reply:
We did need them. You can see them being copied over in the old script here https://github.com/JuliaPackaging/Yggdrasil/pull/2110/files#diff-b829d9f1c7eea63e81e0d3ba703fd8e8837f81501383b544c3343cc10ce977a0L85

Reply:
Current `HDF5_jll` doesn't depend on curl nor openssl:
```
shell> ldd $(HDF5_jll.libhdf5_path)
        linux-vdso.so.1 (0x00007ffe803c4000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007fcdb915c000)
        libsz-1c7dd0cf.so.2.0.1 => /home/mose/.julia/artifacts/37f94a8b66cdbf26fcb2defdbc6c2b147f2145f2/lib/./libsz-1c7dd0cf.so.2.0.1 (0x00007fcdb8f57000)
        libaec-2147abcd.so.0.0.4 => /home/mose/.julia/artifacts/37f94a8b66cdbf26fcb2defdbc6c2b147f2145f2/lib/./libaec-2147abcd.so.0.0.4 (0x00007fcdb8d4e000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007fcdb8d34000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007fcdb8d2e000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007fcdb8be8000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007fcdb8a1d000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007fcdb89fb000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007fcdb97b6000)
shell> ldd $(HDF5_jll.libhdf5_hl_path)
        linux-vdso.so.1 (0x00007ffc5c78f000)
        libhdf5-774513bb.so.103.0.0 => /home/mose/.julia/artifacts/37f94a8b66cdbf26fcb2defdbc6c2b147f2145f2/lib/./libhdf5-774513bb.so.103.0.0 (0x00007f9fb41a6000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007f9fb415e000)
        libsz-1c7dd0cf.so.2.0.1 => /home/mose/.julia/artifacts/37f94a8b66cdbf26fcb2defdbc6c2b147f2145f2/lib/./libsz-1c7dd0cf.so.2.0.1 (0x00007f9fb3f59000)
        libaec-2147abcd.so.0.0.4 => /home/mose/.julia/artifacts/37f94a8b66cdbf26fcb2defdbc6c2b147f2145f2/lib/./libaec-2147abcd.so.0.0.4 (0x00007f9fb3d50000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f9fb3d36000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f9fb3d30000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f9fb3be8000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f9fb3a1f000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f9fb39fd000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f9fb49e8000)
```
So, did anything change?

Reply:
Is this due to the PyPi -> Conda move?

Reply:
I still don't see libcurl though:
```
% ldd *.so
libhdf5_hl.so:
        linux-vdso.so.1 (0x00007ffd50bc3000)
        libhdf5.so.200 => /tmp/mozilla_mose0/x86_64-linux-gnu/lib/././libhdf5.so.200 (0x00007f10bddcd000)
        libcrypto.so.1.1 => /usr/lib/libcrypto.so.1.1 (0x00007f10bdab2000)
        libcurl.so.4 => /usr/lib/libcurl.so.4 (0x00007f10bda1c000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007f10bda11000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f10bd9ef000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f10bd9d5000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f10bd9cd000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f10bd887000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f10bd6be000)
        libnghttp2.so.14 => /usr/lib/libnghttp2.so.14 (0x00007f10bd692000)
        libidn2.so.0 => /usr/lib/libidn2.so.0 (0x00007f10bd671000)
        libssh2.so.1 => /usr/lib/libssh2.so.1 (0x00007f10bd631000)
        libpsl.so.5 => /usr/lib/libpsl.so.5 (0x00007f10bd61c000)
        libssl.so.1.1 => /usr/lib/libssl.so.1.1 (0x00007f10bd58c000)
        libgssapi_krb5.so.2 => /usr/lib/libgssapi_krb5.so.2 (0x00007f10bd537000)
        libkrb5.so.3 => /usr/lib/libkrb5.so.3 (0x00007f10bd451000)
        libk5crypto.so.3 => /usr/lib/libk5crypto.so.3 (0x00007f10bd420000)
        libcom_err.so.2 => /usr/lib/libcom_err.so.2 (0x00007f10bd41a000)
        libzstd.so.1 => /usr/lib/libzstd.so.1 (0x00007f10bd346000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f10be1f3000)
        libunistring.so.2 => /usr/lib/libunistring.so.2 (0x00007f10bd1c4000)
        libkrb5support.so.0 => /usr/lib/libkrb5support.so.0 (0x00007f10bd1b5000)
        libkeyutils.so.1 => /usr/lib/libkeyutils.so.1 (0x00007f10bd1ae000)
        libresolv.so.2 => /usr/lib/libresolv.so.2 (0x00007f10bd194000)
libhdf5.so:
        linux-vdso.so.1 (0x00007fff337e1000)
        libcrypto.so.1.1 => /usr/lib/libcrypto.so.1.1 (0x00007f62a2361000)
        libcurl.so.4 => /usr/lib/libcurl.so.4 (0x00007f62a22cb000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007f62a22c0000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f62a229e000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f62a2284000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f62a227e000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f62a2136000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f62a1f6d000)
        libnghttp2.so.14 => /usr/lib/libnghttp2.so.14 (0x00007f62a1f41000)
        libidn2.so.0 => /usr/lib/libidn2.so.0 (0x00007f62a1f20000)
        libssh2.so.1 => /usr/lib/libssh2.so.1 (0x00007f62a1ee0000)
        libpsl.so.5 => /usr/lib/libpsl.so.5 (0x00007f62a1ecd000)
        libssl.so.1.1 => /usr/lib/libssl.so.1.1 (0x00007f62a1e3b000)
        libgssapi_krb5.so.2 => /usr/lib/libgssapi_krb5.so.2 (0x00007f62a1de6000)
        libkrb5.so.3 => /usr/lib/libkrb5.so.3 (0x00007f62a1d00000)
        libk5crypto.so.3 => /usr/lib/libk5crypto.so.3 (0x00007f62a1ccf000)
        libcom_err.so.2 => /usr/lib/libcom_err.so.2 (0x00007f62a1cc9000)
        libzstd.so.1 => /usr/lib/libzstd.so.1 (0x00007f62a1bf7000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f62a2a7c000)
        libunistring.so.2 => /usr/lib/libunistring.so.2 (0x00007f62a1a73000)
        libkrb5support.so.0 => /usr/lib/libkrb5support.so.0 (0x00007f62a1a64000)
        libkeyutils.so.1 => /usr/lib/libkeyutils.so.1 (0x00007f62a1a5d000)
        libresolv.so.2 => /usr/lib/libresolv.so.2 (0x00007f62a1a43000)
```

Reply:
And this depends on a lot of other libraries not listed here (we don't even have libkrb5 at all as far as I can see)

Reply:
I don't think this can work on any arbitrary system.  @staticfloat why isn't audit catching the missing libraries?

Reply:
I'll double check, but I recall the resolver spitting warnings that these libraries alwere missing, which definitely included even OpenSSL for libcrypt or something I can't recall of the top of my head.

Reply:
> why isn't audit catching the missing libraries?

My first guess is that when we `dlopen()` on a native platform, it picks some libraries up from the greater environment.

We may need to alter the audit pass to first collect the names of each dependency that it looks for, then after we `dlopen()`, we walk over `dllist()` and find each library and ensure that the path of that library is actually coming from the prefix.